tests/helpers.py

```diff
@@ -33,24 +33,28 @@ def step_env_with_gym_quirks(test_case,
         # Roundtrip serialization
         round_trip = pickle.loads(pickle.dumps(env))
         assert round_trip.env.spec == env.env.spec
-        env = round_trip
+        _env = round_trip
+    else:
+        _env = env
 
-    env.reset()
+    _env.reset()
     for _ in range(n):
-        _, _, done, _ = env.step(env.action_space.sample())
+        _, _, done, _ = _env.step(_env.action_space.sample())
         if render:
             if spec.id not in KNOWN_GYM_RENDER_NOT_IMPLEMENTED:
-                env.render()
+                _env.render()
             else:
                 with test_case.assertRaises(NotImplementedError):
-                    env.render()
+                    _env.render()
+            if hasattr(env.env, 'viewer') and env.env.viewer is None:
```

Comment:
this is really hacky and hard to understand. it also makes assumptions about the caller.

is there a reason you can't just close the viewer on the serialized environment here, immediately?

Reply:
I agree. And yeah, closing window on serialized environment makes more sense. I'll update it.